### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by robatipoor**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -83,7 +83,9 @@ fn run() -> Result<(), ContextualError> {
         && miniserve_config
             .path
             .symlink_metadata()
-            .map_err(|e| ContextualError::IOError("Failed to retrieve symlink's metadata".to_string(), e))?
+            .map_err(|e| {
+                ContextualError::IOError("Failed to retrieve symlink's metadata".to_string(), e)
+            })?
```
Comment (KSXGitHub) [inline src/main.rs]: You just split this line into 3, is this necessary?

File: src/themes.rs [hunk 2]
```diff
@@ -17,14 +62,13 @@ impl ColorScheme {
     /// Returns the URL-compatible name of a color scheme
     /// This must correspond to the name of the variant, in lowercase
     /// See https://github.com/svenstaro/miniserve/pull/55 for explanations
-    pub fn to_slug(self) -> String {
+    pub fn to_slug(self) -> &'static str {
```
Comment (KSXGitHub) [inline src/themes.rs]: I like this change.

File: src/main.rs (hunk 1 above)
Comment (robatipoor) [reply src/main.rs]: rustfmt auto split line to 3

Comment (KSXGitHub) [reply src/main.rs]: It seems that you need to undo that to please Clippy

https://travis-ci.org/svenstaro/miniserve/jobs/538610810

File: src/archive.rs [hunk 3]
```diff
@@ -97,12 +99,12 @@ fn tar(
...
     // Recursively adds the content of src_dir into the archive stream
     tar_builder
-        .append_dir_all(inner_folder, &src_dir)
+        .append_dir_all(inner_folder, src_dir.as_ref())
         .map_err(|e| {
             ContextualError::IOError(
                 format!(
                     "Failed to append the content of {} to the TAR archive",
-                    &src_dir
+                    src_dir.as_ref().to_str().unwrap()
```
Comment (svenstaro) [inline src/archive.rs]: I would prefer not to have this `unwrap()`. Can you choose a sensible default? I know it's very unlikely to fail, but still.

Comment (robatipoor) [reply src/archive.rs]: I have no idea except using `unwrap_or("file")` If you think good, I'll change it

**Comment by svenstaro** [comment]
Very cool, I like these changes! Thanks.

**Pull request merged**